### PR TITLE
Show link to job details page for each multiremote instance - …

### LINKED ISSUE #4645: Sauce Labs Job URL not being created in `spec` Reporter when using Multiremote
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.7.6
 - **Mode:** Using WDIO test runner
 - **If WDIO Testrunner, running sync/async:** async
 - **Node.js version:** v12.2.0
 - **NPM version:** 6.9.0
 - **Browser name and version:** Chrome, latest versions
 - **Platform name and version:** MacOS, Windows 10
 - **Additional wdio packages used (if applicable):** @wdio/spec reporter, @wdio/sauce-service

**Config of WebdriverIO**
Here is my config:

```
exports.config = {
   
    runner: 'local',
   
    user: process.env.SAUCE_USERNAME,
    key: process.env.SAUCE_ACCESS_KEY,
    
    service: ['sauce'],
        specs: [
        'tests/multi*.js'
    ],
    
    maxInstances: 40,
    
    capabilities: {
        
	    myChrome: {capabilities: {browserName: 'chrome', browserVersion: 'latest', platformName: 'Windows 10'}},
        myChrome2: {capabilities: {browserName: 'chrome', browserVersion: 'latest', platformName: 'Windows 10'}},
    },
    
    logLevel: 'silent',
    
    bail: 0,
    baseUrl: 'https://www.saucedemo.com',
    waitforTimeout: 10000,
    

    framework: 'mocha',
  
    reporters: ['dot','spec'],
    
    
    mochaOpts: {
        ui: 'bdd',
        timeout: 60000
    }
}
```

**Describe the bug**
When executing multi-remote sessions on Sauce Labs, the sessions are created and executed without issue. The reporter output however does not provide a link to the Sauce job. 

**Expected output**:
```
[MultiremoteBrowser #0-0] Spec: /Users/joshuagrant/Documents/saucelabs-sample-test-frameworks/JS-Mocha-WebdriverIO-Selenium/tests/multi_login.j
s
[MultiremoteBrowser #0-0] Running: undefined
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Multiuser Login
[MultiremoteBrowser #0-0]    ✓ correct messages appear
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] 1 passing (5.9s)
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Check out job at https://app.saucelabs.com/tests/abc1234567890fgh
```

**Actual output**:
```
[MultiremoteBrowser #0-0] Spec: /Users/joshuagrant/Documents/saucelabs-sample-test-frameworks/JS-Mocha-WebdriverIO-Selenium/tests/multi_login.j
s
[MultiremoteBrowser #0-0] Running: undefined
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Multiuser Login
[MultiremoteBrowser #0-0]    ✓ correct messages appear
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] 1 passing (5.9s)
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Check out job at https://app.saucelabs.com/tests/undefined
```

**To Reproduce**

1. Create wdio.conf.js as above
2. Execute spec that opens two browsers and closes them
3. Read Reporter output on console.

Miraculously, mutli-remote does work well on Sauce Labs, but this bug would be an excellent fixup to round out the functionality.

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

When run multiremote tests on Sauce Labs with the spec reporter, it would not show the links to the job details page correctly:

```
[MultiremoteBrowser #0-0] Spec: /Users/joshuagrant/Documents/saucelabs-sample-test-frameworks/JS-Mocha-WebdriverIO-Selenium/tests/multi_login.j
s
[MultiremoteBrowser #0-0] Running: undefined
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Multiuser Login
[MultiremoteBrowser #0-0]    ✓ correct messages appear
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] 1 passing (5.9s)
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Check out job at https://app.saucelabs.com/tests/undefined
```

This patch fixes that and shows the link including the instance name:

```
[MultiremoteBrowser #0-0] Spec: /foo/bar/baz.js
[MultiremoteBrowser #0-0] Running: MultiremoteBrowser
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Foo test
[MultiremoteBrowser #0-0]    green ✓ foo
[MultiremoteBrowser #0-0]    green ✓ bar
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Bar test
[MultiremoteBrowser #0-0]    green ✓ some test
[MultiremoteBrowser #0-0]    red ✖ a failed test
[MultiremoteBrowser #0-0]    red ✖ a failed test with no stack
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Baz test
[MultiremoteBrowser #0-0]    green ✓ foo bar baz
[MultiremoteBrowser #0-0]    cyan - a skipped test
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] green 4 passing (5s)
[MultiremoteBrowser #0-0] red 1 failing
[MultiremoteBrowser #0-0] cyan 1 skipped
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] 1) Bar test a failed test
[MultiremoteBrowser #0-0] red expected foo to equal bar
[MultiremoteBrowser #0-0] gray Failed test stack trace
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] 2) Bar test a failed test with no stack
[MultiremoteBrowser #0-0] red expected foo to equal bar
[MultiremoteBrowser #0-0]
[MultiremoteBrowser #0-0] Check out browserA job at https://app.saucelabs.com/tests/foobar
[MultiremoteBrowser #0-0] Check out browserB job at https://app.saucelabs.com/tests/barfoo
```

closes #4645

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by mgrybyk** [comment]
Codecov complaining about missing test for https://codecov.io/gh/webdriverio/webdriverio/pull/4833/diff?src=pr&el=tree#diff-cGFja2FnZXMvd2Rpby1ydW5uZXIvc3JjL2luZGV4Lmpz

File: packages/wdio-spec-reporter/src/index.js [hunk 1]
```diff
@@ -75,12 +75,20 @@ class SpecReporter extends WDIOReporter {
...
+                instanceName
+            }))
+            : this.getTestLink(runner)
         const output = [
             ...this.getHeaderDisplay(runner),
             ...results,
             ...this.getCountDisplay(duration),
             ...this.getFailureDisplay(),
-            ...this.getTestLink(runner)
+            ...(testLinks.length ? ['', ...testLinks] : testLinks)
```
Comment (mgrybyk) [inline packages/wdio-spec-reporter/src/index.js]: Maybe it's better to check if testLinks is array here? 

Why the first element is ' '?

**Comment by christian-bromann** [comment]
@mgrybyk I made it a bit more clear .. the empty strings generate a new line in the report. I also added the session id to the report as mentioned in #4834

**Comment by mgrybyk** [comment]
Are you going to deal with coverage https://codecov.io/gh/webdriverio/webdriverio/pull/4833/diff?src=pr&el=tree#diff-cGFja2FnZXMvd2Rpby1ydW5uZXIvc3JjL2luZGV4Lmpz
?

**Comment by christian-bromann** [comment]
I wasn't .. I added a line to a section that isn't currently covered anyway.

**Comment by pmerwin** [comment]
has this been merged in?  Still seeing an issue on latest vs 7.

**Pull request merged**